### PR TITLE
One socket for each module

### PR DESCRIPTION
**Pull request by Mihnea9901**
- Reworked CreateInterface so now every module can create its own socket and keep it alive until module shuts down, so each module is responsible for its socket
- This way, when you send a frame from an ecu, you won't receive the frame you sent, so buffer should be cleaner.
- Added getters in each module so you can access the module's socket.
- Modified RDBI service a bit to accept new socket format and also removed something I missed in the last PR.
- Removed SRCSTEST from createInterface makefile job, because it doesn't need all the files.

Now you don't need to access interface.getInstance in your service because it will be passed as argument when it is called in handleFrames.
Socket will be passed to handleFrames object from the module class, and when service will be called, that socket will be passed as argument.
On Mcu side, I added a method to determine which socket to use.

PR is into DEV_BE as these are common changes and I will merge them back to UDS and OTA.

### CODE REVIEW

**Review by TheoRandstad — APPROVED** [review]
Seems fine for me, need to check UT and to integrate also in securityAccess

**Review by alexandrudoncea — APPROVED** [review]

File: src/ota/request_update_status/include/RequestUpdateStatus.h [hunk 1]
```diff
@@ -26,7 +26,6 @@
 #include <map>
 #include "../../../utils/include/Logger.h"
 // #include "../../../uds/read_data_by_identifier/include/read_data_by_identifier.h"
```
Comment (RubenNatanael) [inline src/ota/request_update_status/include/RequestUpdateStatus.h:28]: Please use /**/ to comment

File: src/ota/request_download/src/RequestDownload.cpp [hunk 2]
```diff
@@ -112,16 +114,12 @@ void RequestDownloadService::requestDownloadRequest(int id, std::vector<uint8_t>
...
     uint8_t frame_dest_id = id & 0xFF;
     LOG_INFO(RDSlogger.GET_LOGGER(), "frame id dest: 0x{0:x}", static_cast<int>(frame_dest_id));
     if(frame_dest_id == 0x10)
     {
         std::lock_guard<std::mutex> lock(create_interface_mutex);
-        create_interface = create_interface ->getInstance(0x00, MCULogger);
-        if (create_interface)
-        {
-            int socket_fd = create_interface->getSocketEcuWrite();
             LOG_INFO(RDSlogger.GET_LOGGER(), "log in service");
```
Comment (RubenNatanael) [inline src/ota/request_download/src/RequestDownload.cpp:123]: Take care with indentation, align lines correctly

**Review by RubenNatanael — APPROVED** [review]

File: src/ota/request_update_status/include/RequestUpdateStatus.h (hunk 1 above)
Comment (Mihnea9901) [reply src/ota/request_update_status/include/RequestUpdateStatus.h]: done

File: src/ota/request_download/src/RequestDownload.cpp (hunk 2 above)
Comment (Mihnea9901) [reply src/ota/request_download/src/RequestDownload.cpp]: done

**Review by TheoRandstad — APPROVED** [review]

**Pull request merged**